sdk/src/main/scala/com/horizen/certificatesubmitter/CertificateSubmitter.scala

```diff
@@ -736,8 +535,15 @@ object CertificateSubmitter {
 object CertificateSubmitterRef {
   def props(settings: SidechainSettings, sidechainNodeViewHolderRef: ActorRef, params: NetworkParams,
             mainchainChannel: MainchainNodeChannel)
-           (implicit ec: ExecutionContext): Props =
-    Props(new CertificateSubmitter(settings, sidechainNodeViewHolderRef, params, mainchainChannel)).withMailbox("akka.actor.deployment.submitter-prio-mailbox")
+           (implicit ec: ExecutionContext): Props = {
+    val keyRotationStrategy = if (settings.withdrawalEpochCertificateSettings.typeOfCircuit  == TypeOfCircuit.NaiveThresholdSignatureCircuit) {
```

Comment:
note: even if settings have knowledge about circuit type, we should use `params`.
Simply all network specific params must be addressed in `NetworkParams`, `SidechainSettings` is just one of the sources for network params.